### PR TITLE
fix: remove dead code, enable re-applying of the stored procedure, remove unused key from stored procedure

### PR DESCRIPTION
**Pull request by prb112**
…move unused key from stored procedure

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/ProcedureDef.java [hunk 1]
```diff
@@ -49,6 +50,8 @@ public void apply(Integer priorVersion, IDatabaseAdapter target) {
         }
 
         // Procedures are applied with "Create or replace", so just do a regular apply
+        // we need to drop and then apply.
```
Comment (lmsurpre) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/ProcedureDef.java:53]: does this comment invalidate the comment on the previous line?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java [hunk 2]
```diff
@@ -464,21 +464,22 @@ protected void updateProcedures() {
 
         // Now only apply the procedures in the model. Much faster than
         // going through the whole schema
-        try {
-            try (Connection c = createConnection()) {
+        try (ITransaction tx = TransactionFactory.openTransaction(connectionPool)) {
```
Comment (lmsurpre) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:467]: I think its fine, but where were we starting the transaction before?

Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:467]: We were not starting a transaction before, we were using a statement which autocommitted the change.

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/ProcedureDef.java [hunk 3]
```diff
@@ -49,6 +50,8 @@ public void apply(Integer priorVersion, IDatabaseAdapter target) {
         }
 
         // Procedures are applied with "Create or replace", so just do a regular apply
```
Comment (prb112) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/ProcedureDef.java:52]:
```suggestion
```

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/ProcedureDef.java (hunk 1 above)
Comment (prb112) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/ProcedureDef.java:52]: Yes, you are right.   Fixed

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:467]: ah, ok.  i remember trying something like this in the past and getting into trouble with the migration code...but I think that was before robin cleaned up all the transaction handling.  tests are passing, so i'm good if @punktilious is

**Review by lmsurpre — APPROVED** [review]
LGTM

**Comment by prb112** [comment]
@punktilious I've merged this PR so we can close Sprint 19.

**Pull request merged**